File: solutions/kotlin/hamming/3/src/main/kotlin/Hamming.kt
import java.lang.IllegalArgumentException

object Hamming {

    fun compute(leftStrand: String, rightStrand: String): Int {

        if (leftStrand.length != rightStrand.length) {
            throw IllegalArgumentException("left and right strands must be of equal length")
        }
        
        return (leftStrand zip rightStrand).count {
            it.first != it.second
        }
    }

}

File: solutions/lua/hamming/3/hamming.lua
local Hamming = {}

local distance = function(a, b)
    local diff = 0
    for i = 1, #a do
        if a:sub(i, i) ~= b:sub(i, i) then
            diff = diff + 1
        end
    end

    return diff
end

function Hamming.compute(a,b)

    local length = #a
    if length ~= #b then
        return -1
    end

    return distance(a, b)
end


return Hamming

